go/vt/sqlparser/ast.go

```diff
@@ -581,6 +581,17 @@ func (node Columns) WalkSubtree(visit Visit) error {
 	return nil
 }
 
+// FindColumn finds a column in the column list, returning
+// the index if it exists or -1 otherwise
+func (node Columns) FindColumn(col ColIdent) int {
```

Comment:
I added this instead of using the schema index like I was before, because I believe that was assuming the person inserts using all the columns in the schema or in the same order of the schema.  This will find the index of the column based on the order of the insert itself.